examples/contracts/generic_contract/src/contract.rs

```diff
@@ -0,0 +1,99 @@
+use cosmwasm_schema::cw_serde;
+use cosmwasm_std::{Reply, Response, StdResult};
+use serde::de::DeserializeOwned;
+use serde::Deserialize;
+use sylvia::types::{CustomMsg, ExecCtx, InstantiateCtx, MigrateCtx, QueryCtx, ReplyCtx};
+use sylvia::{contract, schemars};
+
+#[cw_serde]
+pub struct ExternalMsg;
+impl cosmwasm_std::CustomMsg for ExternalMsg {}
+
+pub struct GenericContract<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType>(
+    std::marker::PhantomData<(
+        InstantiateParam,
+        ExecParam,
+        QueryParam,
+        MigrateParam,
+        RetType,
+    )>,
+);
+
+#[contract]
+impl<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType>
+    GenericContract<InstantiateParam, ExecParam, QueryParam, MigrateParam, RetType>
+where
+    for<'msg_de> InstantiateParam: CustomMsg + Deserialize<'msg_de> + 'msg_de,
+    for<'exec> ExecParam: CustomMsg + DeserializeOwned + 'exec,
+    for<'exec> QueryParam: CustomMsg + DeserializeOwned + 'exec,
+    for<'exec> MigrateParam: CustomMsg + DeserializeOwned + 'exec,
+    for<'ret> RetType: CustomMsg + DeserializeOwned + 'ret,
+{
+    pub const fn new() -> Self {
+        Self(std::marker::PhantomData)
+    }
+
+    #[msg(instantiate)]
+    pub fn instantiate(&self, _ctx: InstantiateCtx, _msg: InstantiateParam) -> StdResult<Response> {
+        Ok(Response::new())
+    }
+
+    #[msg(exec)]
+    pub fn execute(&self, _ctx: ExecCtx, _msg: ExecParam) -> StdResult<Response> {
+        Ok(Response::new())
+    }
+
+    #[msg(query)]
+    pub fn query(&self, _ctx: QueryCtx, _msg: QueryParam) -> StdResult<Response> {
+        Ok(Response::new())
+    }
+
+    #[msg(migrate)]
+    pub fn migrate(&self, _ctx: MigrateCtx, _msg: MigrateParam) -> StdResult<Response> {
+        Ok(Response::new())
+    }
+
+    #[allow(dead_code)]
+    #[msg(reply)]
+    fn reply(&self, _ctx: ReplyCtx, _reply: Reply) -> StdResult<Response> {
+        Ok(Response::new())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use cosmwasm_std::Empty;
+    use sylvia::multitest::App;
+
+    use crate::contract::ExternalMsg;
+
+    #[test]
+    fn generic_contract() {
+        use super::multitest_utils::CodeId;
+        let app = App::default();
+        let code_id: CodeId<
+            cw_multi_test::BasicApp<Empty, Empty>,
+            ExternalMsg,
+            ExternalMsg,
+            ExternalMsg,
+            super::ExternalMsg,
+            super::ExternalMsg,
+        > = CodeId::store_code(&app);
```

Comment:
Do we really need to pass all those generics? Especially the `BaxicApp` - cannot it be `_`? The rest are fairly okay as those are just contract types.
Also, I think the `App` should be the last generic - so the user can completely ignore it as it should be elideable.

Reply:
Unfortunately they are required as `CodeId` is generic over types defined on contract.

We can omit the `App` though and I think it's a good idea to move it to last argument.